src/main/java/core/basesyntax/strategy/impl/ApplyStrategyImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.exception.FruitStoreException;
+import core.basesyntax.model.FruitNegotiation;
+import core.basesyntax.strategy.ApplyStrategy;
+
+public class ApplyStrategyImpl implements ApplyStrategy {
+    @Override
+    public void process(FruitNegotiation fruit) {
+        switch (fruit.getOperation()) {
+            case BALANCE:
+                new ApplyBalance(fruit).apply();
+                break;
+            case PURCHASE:
+                new ApplyPurchase(fruit).apply();
+                break;
+            case RETURN:
+                new ApplyReturn(fruit).apply();
+                break;
+            case SUPPLY:
+                new ApplySupplay(fruit).apply();
+                break;
+            default:
+                throw new FruitStoreException("Missing type of operation " + fruit.getOperation());
+        }
+    }
```

Comment:
strategy is made exactly to avoid the switch statement as a more convenient replacement for it, so let's get use of that by removing the switch and having a map of `<Operation, UnaryOperation>` instead to get the handler by the operation and use it 
\+ would be nice to accept it as a parameter in a constructor and a strategy itself to also be accepted as a parameter in the `FruitTransactionEvaluator` constructor

Reply:
refactored